src/components/Delegates/DelegateCardList/DelegateTable.tsx

```diff
@@ -0,0 +1,108 @@
+"use client";
+
+import { useEffect, useRef, useState } from "react";
+import InfiniteScroll from "react-infinite-scroller";
+import { DialogProvider } from "@/components/Dialogs/DialogProvider/DialogProvider";
+import { DelegateChunk } from "@/app/api/common/delegates/delegate";
+import { useAgoraContext } from "@/contexts/AgoraContext";
+import { PaginatedResult, PaginationParams } from "@/app/lib/pagination";
+import {
+  Table,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+import DelegateTableRow from "./DelegateTableRow";
+
+interface Props {
+  initialDelegates: PaginatedResult<DelegateChunk[]>;
+  fetchDelegates: (
+    pagination: PaginationParams,
+    seed?: number
+  ) => Promise<PaginatedResult<DelegateChunk[]>>;
+}
+
+export default function DelegateTable({
+  initialDelegates,
+  fetchDelegates,
+}: Props) {
+  const [meta, setMeta] = useState(initialDelegates.meta);
+  const [delegates, setDelegates] = useState(initialDelegates.data);
+
+  const fetching = useRef(false);
+
+  const { setIsDelegatesFiltering } = useAgoraContext();
+
+  useEffect(() => {
+    setIsDelegatesFiltering(false);
+    setDelegates(initialDelegates.data);
+    setMeta(initialDelegates.meta);
+  }, [initialDelegates, setIsDelegatesFiltering]);
+
+  const loadMore = async () => {
+    if (!fetching.current && meta.has_next) {
+      fetching.current = true;
+      const data = await fetchDelegates(
+        { offset: meta.next_offset, limit: meta.total_returned },
+        initialDelegates.seed || Math.random()
+      );
+      setDelegates(delegates.concat(data.data));
+      setMeta(data.meta);
+      fetching.current = false;
+    }
+  };
+
+  return (
+    <DialogProvider>
+      <Table className="min-w-full">
+        <TableHeader className="text-xs text-secondary sticky top-0 bg-white z-10">
+          <TableRow>
```

Comment:
it'd be nice if clicking the row took you to the delegate profile, the same way clicking the card takes you to delegate profile in grid view